### PR TITLE
Refactored runtime.Object helper functions into subpkg

### PR DESCRIPTION
**Pull request by alejandrox1**
Organized functions that abstract the access of
k8s.io/apimachinery/pkg/runtime.Objects into a framework subpackge.

Signed-off-by: Jorge Alarcon Ochoa <alarcj137@gmail.com>

**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Follow up of #77526, it moves various functions for handling runtime.Objects from `e2e/framework/pod` to its own pkg `e2e/framework/runtimeobject`.

**Which issue(s) this PR fixes**:
Part of #77095

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

/area e2e-test-framework
/area test
/sig testing
/priority backlog

### CODE REVIEW

**Comment by alejandrox1** [comment]
ptal
/assign @andrewsykim @timothysc @neolit123

File: test/e2e/framework/runtimeobject/runtimeobject.go [hunk 1]
```diff
@@ -34,9 +34,9 @@ import (
 	extensionsinternal "k8s.io/kubernetes/pkg/apis/extensions"
 )
 
-// TODO: This function is generic enough and used enough that it should be
-// moved to its own subpkg.
-func getRuntimeObjectForKind(c clientset.Interface, kind schema.GroupKind, ns, name string) (runtime.Object, error) {
+// GetRuntimeObjectForKind returns a runtime.Object based on its name, kind,
```
Comment (neolit123) [inline test/e2e/framework/runtimeobject/runtimeobject.go]: > GetRuntimeObjectForKind returns a runtime.Object based on its GroupKind, namespace and name.

**Review by neolit123 — COMMENTED** [review]
thanks @alejandrox1 
added one minor comment.

/approve

File: test/e2e/framework/runtimeobject/runtimeobject.go (hunk 1 above)
Comment (alejandrox1) [reply test/e2e/framework/runtimeobject/runtimeobject.go]: Thank you for your comment! Fixed it

**Comment by neolit123** [comment]
/lgtm
/hold
for other to see too.

**Comment by spiffxp** [comment]
This package isn't called out in the description of https://github.com/kubernetes/kubernetes/issues/77095, I notice there was a `resource/runtime.go` called out in that issue, is that something different?

**Comment by alejandrox1** [comment]
> This package isn't called out in the description of #77095, I notice there was a `resource/runtime.go` called out in that issue, is that something different?

ha... i think you are right. i think that it may actually be the same thing.
@andrewsykim  @draveness could you confirm, there isn't much detail on the tracking issue about what would go into resource/runtime.go?

**Comment by andrewsykim** [comment]
Yes `resource/runtime.go` was for runtime object and related functions. `runtimeobject` (or maybe just `runtimeobj`) is probably better at this point so it's not confused with other terms that have `runtime` in it.

**Comment by alejandrox1** [comment]
Lat push was to rename things, i.e., `resource/runtimeobj.go` instead of `runtimeobj/runtimeobj.go`.
What do you all think?

**Comment by spiffxp** [comment]
/lgtm
I leave it to @draveness @andrewsykim or @timothysc to remove the /hold if they agree with the naming

**Comment by draveness** [comment]
> Lat push was to rename things, i.e., `resource/runtimeobj.go` instead of `runtimeobj/runtimeobj.go`.
> What do you all think?

SGTM

File: test/e2e/framework/resource/runtimeobj.go [hunk 2]
```diff
@@ -14,7 +14,7 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package pod
+package runtimeobject
```
Comment (andrewsykim) [inline test/e2e/framework/resource/runtimeobj.go]: Should this be `resource`?

Comment (alejandrox1) [reply test/e2e/framework/resource/runtimeobj.go]: oh yes, it should! just fixed it

**Comment by spiffxp** [comment]
needs rebase but otherwise lgtm

**Comment by alejandrox1** [comment]
pull-kubernetes-bazel-build got stuck with Pod pending timeout. https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/79150/pull-kubernetes-bazel-build/1141884596158205952

/test pull-kubernetes-bazel-build

**Pull request merged**